logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/client/UpdateSelfClientCapabilityToConsumableNotificationsUseCase.kt

```diff
@@ -0,0 +1,128 @@
+    suspend operator fun invoke(): Either<CoreFailure, Unit>
+}
+
+@Suppress("LongParameterList")
+internal class UpdateSelfClientCapabilityToConsumableNotificationsUseCaseImpl internal constructor(
+    private val selfClientIdProvider: CurrentClientIdProvider,
+    private val clientRepository: ClientRepository,
+    private val clientRemoteRepository: ClientRemoteRepository,
+    private val incrementalSyncRepository: IncrementalSyncRepository,
+    private val selfServerConfig: SelfServerConfigUseCase,
+    private val syncRequester: suspend () -> Either<CoreFailure, Unit>,
+    private val slowSyncRepository: SlowSyncRepository,
+) : UpdateSelfClientCapabilityToConsumableNotificationsUseCase {
+
+    private val logger = kaliumLogger.withTextTag("UpdateSelfClientCapabilityToConsumableNotificationsUseCase")
+
+    override suspend fun invoke(): Either<CoreFailure, Unit> {
+        incrementalSyncRepository.incrementalSyncState.filter { it is IncrementalSyncStatus.Live }.collect {
+            val clientHasConsumableNotifications = clientRepository.observeClientHasConsumableNotifications().firstOrNull() ?: false
+            val shouldUpdateCapability =
+                clientHasConsumableNotifications.not() && clientRepository.shouldUpdateClientConsumableNotificationsCapability()
+            val isEnabledByServer =
+                (selfServerConfig() as? SelfServerConfigUseCase.Result.Success)?.serverLinks?.metaData?.commonApiVersion?.version
+                    ?.let { serverApiVersion -> serverApiVersion >= MIN_API_VERSION_FOR_CONSUMABLE_NOTIFICATIONS } ?: false
+
+            if (!shouldUpdateCapability || !isEnabledByServer) {
+                logger.d(
+                    "Skipping client upgrade: shouldUpdateCapability: $shouldUpdateCapability, isEnabledByServer: " +
+                            "$isEnabledByServer, clientHasConsumableNotifications: $clientHasConsumableNotifications"
+                )
+                return@collect
+            }
+            logger.d(
+                "Starting client upgrade: shouldUpdateCapability: $shouldUpdateCapability, isEnabledByServer: " +
+                        "$isEnabledByServer, clientHasConsumableNotifications: $clientHasConsumableNotifications"
+            )
+            performClientCapabilityUpgrade()
+        }
+        return Unit.right()
+    }
+
+    /**
+     * Performs the client capability upgrade to consumable notifications.
+     * After updating, it will execute a quick sync to ensure the client has the latest data.
+     */
+    private suspend fun performClientCapabilityUpgrade() {
+        selfClientIdProvider().flatMap { clientId ->
+            clientRemoteRepository.updateClientCapabilities(
+                updateClientCapabilitiesParam = UpdateClientCapabilitiesParam(
+                    capabilities = listOf(
+                        ClientCapability.LegalHoldImplicitConsent,
+                        ClientCapability.ConsumableNotifications
```

Comment:
I wouldn't expect to see LegalHold in here... But I cancan't think of solutions that wouldn't overengineer this small issue. Could you maybe add a comment here, explaining why LegalHold capability is being added?